### ISSUE TITLE
Support for schema version 18.0.0 on AWS CDK v2

### ISSUE THREAD

**Issue by MariusDK** [aws/aws-cdk#20135]
### Describe the feature

After an upgrade of aws-cdk to 1.154.0 the config files created by cdk synth are based on schema version 18.0.0. 
Building them on a pipeline that uses aws-cdk 2.22.0 the new schema isn’t supposed.

### Use Case

Support AWS cli v2 for schema 18.0.0

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2

### Environment details (OS name and version, etc.)

any env, tested on MacOS also

**Comment by nemmison** [comment]
Been having this issue since yesterday. I keep getting this error:

This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version.
(Cloud assembly schema version mismatch: Maximum schema version supported is 17.0.0, but found 18.0.0)

I'm assuming this is the same issue?

This is quite a big blocker for me, would we have any estimate on how long it's going to be before a fix?

**Comment by estrudels** [comment]
I am also encountering the issue when I upgrade from using **@aws-cdk/....** version `1.153.0` to `1.154.0` 

```
This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version.
(Cloud assembly schema version mismatch: Maximum schema version supported is 17.0.0, but found 18.0.0)
```

**Comment by frumania** [comment]
As a workaround, this seems to work
`npm i -g aws-cdk@latest-1`

**Comment by adriantaut** [comment]
facing the same issue while using `1.154.0` dependencies. Locally I have `aws-cdk@2.22.0`

**Comment by shellscape** [comment]
This is horrible mismanagement of versioning by the folks at AWS.

- There's no mention of this in the `aws-cdk` README.
- There's no CHANGELOG at https://github.com/aws/aws-cdk/tree/master/packages/aws-cdk to cover the new `dist-tag`
- There's no note on the release https://github.com/aws/aws-cdk/releases/tag/v2.22.0

Poor communication. Poor documentation. I don't know what's going on at AWS lately but between the mismanagement of the `aws-lambda-base-images` and this nonsense, someone in management needs to get the Node teams shaped up.

**Comment by estrudels** [comment]
This should be now fixed with the latest aws-cdk `2.23.0`, it works for me now :)

* https://www.npmjs.com/package/aws-cdk
* https://github.com/aws/aws-cdk/pull/19773
* https://github.com/aws/aws-cdk/releases

**Comment by indrora** [comment]
If there's still customers seeing this issue, please say so now. It looks to have been resolved in a later release (2.23), so I'm putting this under autoclose.
